src/components/UserInfo/UserInfo.tsx

```diff
@@ -1 +1,22 @@
-export const UserInfo = () => {};
+import React from 'react';
+
+type User = {
+  email: string,
+  name: string;
+};
+
+type Props = {
+  user: User | null;
+};
+
+export const UserInfo: React.FC<Props> = ({ user }) => {
+  if (!user) {
+    return null;
+  }
```

Comment:
let's remove this logic from component, UserInfo targets to render the user information, let's not complicate it with the existing logic